src/backend/metal/src/device.rs

```diff
@@ -860,10 +859,69 @@ impl hal::Device<Backend> for Device {
         ID: IntoIterator,
         ID::Item: Borrow<pass::SubpassDependency>,
     {
+        let attachments: Vec<pass::Attachment> = attachments
+            .into_iter()
+            .map(|at| at.borrow().clone())
+            .collect();
+
+        let mut subpasses: Vec<n::Subpass> = subpasses
+            .into_iter()
+            .map(|sp| {
+                let sub = sp.borrow();
+                n::Subpass {
+                    colors: sub.colors
+                        .iter()
+                        .map(|&(id, _)| (id, n::SubpassOps::empty()))
+                        .collect(),
+                    depth_stencil: sub.depth_stencil
+                        .map(|&(id, _)| (id, n::SubpassOps::empty())),
+                    inputs: sub.inputs
+                        .iter()
+                        .map(|&(id, _)| id)
+                        .collect(),
+                    target_formats: n::SubpassFormats {
+                        colors: sub.colors
+                            .iter()
+                            .map(|&(id, _)| {
+                                let format = attachments[id].format.expect("No color format provided");
+                                let mtl_format = self.private_caps.map_format(format).expect("Unable to map color format!");
+                                (mtl_format, Channel::from(format.base_format().1))
+                            })
+                            .collect(),
+                        depth_stencil: sub.depth_stencil
+                            .map(|&(id, _)| {
+                                self.private_caps.map_format(
+                                    attachments[id].format.expect("No depth-stencil format provided")
+                                ).expect("Unable to map depth-stencil format!")
+                            }),
+                    },
+                }
+            })
+            .collect();
+
+        // sprinkle load operations
```

Comment:
This makes sense, but might be useful to expand to the comments here a bit just to explain the intent. Also where do we distinguish between `loadOp` and `stencilLoadOp` (and store) for mixed depth stencil attachments?